### ISSUE TITLE
tomlkit incorrectly parses pyproject.toml

### ISSUE THREAD

**Issue by knowsuchagency** [sdispater/poetry#374]
- [x ] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [ x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [ x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: MacOS High Sierra 10.13.6
- **Poetry version**: 0.11.4
- **Link of a [Gist](https://gist.github.com/knowsuchagency/2e4628537d791dd6b0deec7153790500) with the contents of your pyproject.toml file**: 

## Issue

The tomlkit library incorrectly parses table arrays in the `tool` namespace. The above-mentioned `pyproject.toml` should produce the following json:

```json
{
  "tool": {
    "poetry": {
      "name": "blogify",
      "version": "0.1.0",
      "description": "",
      "authors": [
        "Stephan Fitzpatrick <knowsuchagency@gmail.com>"
      ],
      "dependencies": {
        "python": "*",
        "shell_utils": "^0.6.0",
        "jupyterlab": "^0.33.6",
        "watchdog": "^0.8.3",
        "toml": "^0.9.4",
        "tomlkit": "^0.4.0"
      },
      "dev-dependencies": {
        "pytest": "^3.0"
      }
    },
    "foo": [
      {
        "name": "first"
      },
      {
        "name": "second"
      }
    ]
  }
}
```

tomlkit deserializes it into the following structure

```
{'tool': [{'foo': [{'name': 'second'}]}],
 'tool': [{'foo': [{'name': 'second'}]}]}
```

**Comment by nchammas** [comment]
If this issue is restricted to TOML Kit and not specific to Poetry, shouldn't you file this over on [TOML Kit's issue tracker](https://github.com/sdispater/tomlkit/issues)?

**Comment by knowsuchagency** [comment]
Perhaps, but I figured it would have more visibility here. I don't know that any other library authors are using tomlkit

**Comment by sdispater** [comment]
I just released `tomlkit` 0.4.1 which should fix the issue.
